Modules/IO/ImageBase/include/itkImageSeriesReader.hxx

```diff
@@ -408,16 +413,54 @@ ImageSeriesReader<TOutputImage>::GenerateData()
         ImageAlgorithm::Copy(readerOutput, output, sliceRegionToRequest, outRegion);
       }
 
+      // verify that slice spacing is the expected one
+      // since we can be skipping some slices because they are outside of requested region
+      // I am using additional variable
+      if (prev_slice_is_valid)
+      {
+        typename TOutputImage::PointType slice_origin = readerOutput->GetOrigin();
+        using SpacingScalarType = typename TOutputImage::SpacingValueType;
+        Vector<SpacingScalarType, TOutputImage::ImageDimension> dirN;
+        for (size_t j = 0; j < TOutputImage::ImageDimension; ++j)
+        {
+          dirN[j] =
+            static_cast<SpacingScalarType>(slice_origin[j]) - static_cast<SpacingScalarType>(prev_slice_origin[j]);
+        }
+        SpacingScalarType dirNnorm = dirN.GetNorm();
+
+        if (!Math::AlmostEquals(
+              dirNnorm,
+              output_spacing[this->m_NumberOfDimensionsInImage])) // either non-uniform sampling or missing slice
+        {
+          non_uniform_sampling = true;
+          itkWarningMacro("Non unform sampling or missing slices detected ")
```

Comment:
Is there a missing semi-colon?

Reply:
itkWarningMacro : https://itk.org/Doxygen/html/itkMacro_8h_source.html - should work without semi-colon

Reply:
The ITK code base seems to always treat them like functions though and end with a semi-colon.

itkWarningMacro should probably be reformulated in a do-while loop:

https://wiki.sei.cmu.edu/confluence/display/c/PRE10-C.+Wrap+multistatement+macros+in+a+do-while+loop

Reply:
I agree it works the next line is indented which confuses me

It's like a missing periods.